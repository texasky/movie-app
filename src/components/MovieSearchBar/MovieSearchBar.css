.movie-search-bar input {
    border-radius: 3px;
    border: none;
    height: 30px;
    max-width: 200px;
    padding: 5px 10px;
}