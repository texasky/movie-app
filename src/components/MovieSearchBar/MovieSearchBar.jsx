import { useDispatch, useSelector } from 'react-redux';
import { updateSearchQuery } from '../../state/action-creator/index'

const MovieSearchBar = (props) => {
    const dispatch = useDispatch();
    const searchQuery = useSelector(state => state.moviesStore.searchQuery);

    const setSearchQuery = (event) => {
        dispatch(updateSearchQuery(event.target.value));
    }

    return (
        <input type="text" placeholder='Type to search' value={searchQuery} onChange={(event => setSearchQuery(event))}></input>
    )
}

export default MovieSearchBar;