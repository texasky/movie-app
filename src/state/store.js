import { applyMiddleware, createStore } from "redux"
import configureStore from 'redux-mock-store'
import reducers from "./reducers/index"
import thunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";

const store = createStore(reducers, composeWithDevTools(applyMiddleware(thunk)));

export default store; 