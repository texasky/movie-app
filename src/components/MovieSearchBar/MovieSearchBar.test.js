import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from './../../state/store'
import MovieSearchBar from './MovieSearchBar';

test('Movie search bar component tests', () => {
    const utils = render(<Provider store={store}><MovieSearchBar /></Provider>);
    const input = utils.getByTestId('movie-search-input');

    expect(input).toBeInTheDocument();

    fireEvent.change(input, {target: {value: 'Star Wars'}})
    expect(input.value).toBe('Star Wars')

    fireEvent.change(input, {target: {value: '123'}})
    expect(input.value).toBe('123')

    fireEvent.change(input, {target: {value: ''}})
    expect(input.value).toBe('')
});

test('Movie search bar component matches snapshot', () => {
    const tree = renderer.create(<Provider store={store}><MovieSearchBar /></Provider>).toJSON();
    expect(tree).toMatchSnapshot();
});