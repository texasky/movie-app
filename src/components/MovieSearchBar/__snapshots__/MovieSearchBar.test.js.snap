// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie search bar component matches snapshot 1`] = `
<div
  className="movie-search-bar"
>
  <input
    aria-label="search-input"
    data-testid="movie-search-input"
    disabled={false}
    id="movie-search-input"
    onChange={[Function]}
    placeholder="Type to search"
    type="text"
    value=""
  />
</div>
`;
